pandas/core/generic.py

```diff
@@ -8969,18 +8969,21 @@ def _find_valid_index(self, how):
             is_valid = is_valid.any(1)  # reduce axis 1
 
         if how == 'first':
-            # First valid value case
-            i = is_valid.idxmax()
-            if not is_valid[i]:
-                return None
-            return i
-
-        elif how == 'last':
-            # Last valid value case
-            i = is_valid.values[::-1].argmax()
-            if not is_valid.iat[len(self) - i - 1]:
-                return None
-            return self.index[len(self) - i - 1]
+            idx = is_valid.idxmax()
+            if isinstance(is_valid[idx], ABCSeries):
```

Comment:
what are you trying to do here?

Reply:
Thanks - this block is supposed to check that for multiple counts of same index, at least one is not NA.

However, while testing this with following data, the expected output is not being returned 
```python
x = pd.DataFrame({'b': [1,np.NaN,3]}, index=[1,1,2])
```
Expected `1`, returned `None`

I'll rework this patch and commit again - Thanks again for the question prompt, it was fallacy of assumption on my part (had not checked explicitly for NaN value among the multiple index)

Reply:
the loop is incorrect leading to an error, not sure what I was thinking earlier :) - fixed now and committing

Reply:
Fixed - rebased and committed

Reply:
still not clear on the logic here, why can't this be a mirror of the 'last' logic?